src/rpc/misc.cpp

```diff
@@ -898,18 +899,18 @@ UniValue getaddressdeltas(const UniValue& params, bool fHelp)
     if (!(includeChainInfo && start > 0 && end > 0)) {
         return deltas;
     }
-    LOCK(cs_main);
-
-    CBlockIndex* startIndex = chainActive[start];
-    CBlockIndex* endIndex = chainActive[end];
 
     UniValue startInfo(UniValue::VOBJ);
     UniValue endInfo(UniValue::VOBJ);
-
-    startInfo.push_back(Pair("hash", startIndex->GetBlockHash().GetHex()));
+    {
+        LOCK(cs_main);  // for chainActive
+        if (start > chainActive.Height() || end > chainActive.Height()) {
```

Comment:
AFAICT this conditional will never be reached, because `getHeightRange()` (which is called in this function to set `start` and `end`) contains the same check. Not clear to me from this commit which check is now redundant (i.e. whether `getHeightRange()` is used elsewhere).

Reply:
My thinking here is that the return value of `chainActive.Height()` may be different from when it was sampled from within the call to `getHeightRange()` since `cs_main` was released in the meantime. I wanted to make sure that `start` and `end`, when used as indices to `chainActive` immediately after this check:
```
startInfo.push_back(Pair("hash", chainActive[start]->GetBlockHash().GetHex()));
endInfo.push_back(Pair("hash", chainActive[end]->GetBlockHash().GetHex()));
```
are valid. Do I have an incorrect understanding?

Reply:
No, that's accurate. I was imprecise in my earlier comment: the check is redundant except if a reorg happens in parallel with this command executing (which is the only way that `chainActive.Height()` could decrease between the two checks). Even then, I'm not sure it could be triggered particularly easily, as the reorg itself will hold the main lock, and for most small reorgs the result is going to be at least the same height (in order for the reorg to be towards a chain with more work). But I agree that this eliminates the possibility, so I'm happy to leave it in.